src/model/timeLineItems.model.ts

```diff
@@ -0,0 +1,5 @@
+export interface TimeLineItem {
+  name: string
```

Comment:
IMHO, `name` is not the correct key. Probably, it's better to create a function for TimeLineItem like `toString` to return the string to be shown since we have both year and semester.